drone/src/drone/agent/mod.rs

```diff
@@ -132,24 +154,19 @@ pub async fn run_agent(agent_opts: AgentOptions) -> Result<()> {
             cluster: cluster.clone(),
             ip,
         };
-        do_with_retry(
-            || nats.request(&request),
-            30,
-            Duration::from_secs(10),
-        )
-        .await?
+        do_with_retry(|| nats.request(&request), 30, Duration::from_secs(10)).await?
     };
 
     match result {
         DroneConnectResponse::Success { drone_id } => {
-            {
-                let nats = nats.clone();
-                let cluster = cluster.clone();
-                tokio::spawn(ready_loop(nats, drone_id, cluster));
-            }
-
+            let executor = Arc::new(Executor::new(docker, db, nats.clone()));
+            let result = tokio::try_join!(
+                ready_loop(nats.clone(), drone_id, cluster.clone()),
+                listen_for_spawn_requests(drone_id, executor.clone(), nats.clone()),
+                listen_for_termination_requests(executor.clone(), nats.clone())
+            );
             tracing::info!("Listening for spawn requests.");
```

Comment:
This won't be run until after the `try_join`, should move it above.

Reply:
moved it into listen_for_spawn_requests. might be a good idea to put these Listening for {{X}} traces into the nats subscribe call.